### PR TITLE
probably fixes pandoras attacking their owners

### LINKED ISSUE #10311: pandora elite can still hurt its owner using its attacks
## Reproduction:
vortex blasts still hurty

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

Probably stops pandoras from attacking their owners. Closes https://github.com/Citadel-Station-13/Citadel-Station-13/issues/10311

## Why It's Good For The Game

Because they aren't supposed to attack their owners?

## Changelog
:cl:
fix: pandoras attacking their owners
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm [hunk 1]
```diff
@@ -594,12 +594,16 @@ Difficulty: Normal
...
 
-/obj/effect/temp_visual/hierophant/blast/Initialize(mapload, new_caster, friendly_fire, list/only_hit_once)
+/obj/effect/temp_visual/hierophant/blast/Initialize(mapload, new_caster, friendly_fire, list/only_hit_once, list/donthurt)
 	. = ..()
 	if(only_hit_once)
 		hit_things = only_hit_once
 	friendly_fire_check = friendly_fire
+	if (!donthurt)
+		donthurt = list()
+	nohurt = donthurt
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm]: Do this:
```
	if(donthurt)
		hit_things += donthurt
```

File: code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm [hunk 2]
```diff
@@ -627,6 +631,8 @@ Difficulty: Normal
 	if(!damage)
 		return
 	for(var/mob/living/L in T.contents - hit_things) //find and damage mobs...
+		if (L in nohurt)
+			continue
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm]: Remove this.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm (hunk 2 above)
Comment (Seris02) [reply code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm]: ...but then it does nothing.

Comment (Seris02) [reply code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm]: nevermind, I see

**Comment by Seris02** [comment]
oh.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**